File: src/main/java/com/github/cannor147/model/GeoMapDto.java
package com.github.cannor147.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.github.cannor147.namer.Synonymized;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class GeoMapDto implements Synonymized {
    private String name;
    private String[] synonyms;
    @JsonProperty("data")
    private String[] dataFilePaths;
    @JsonProperty("map")
    private String mapFilePath;
    @JsonProperty("background")
    private String backgroundFilePath;
}

File: src/main/java/com/github/cannor147/geomapper/GeoMapDto.kt
@file:Suppress("ArrayInDataClass")

package com.github.cannor147.geomapper

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.cannor147.geomapper.namer.Synonymized

data class GeoMapDto(
    @JsonProperty("name") override val name: String,
    @JsonProperty("synonyms") override val synonyms: Array<String>,
    @JsonProperty("data") val dataFilePaths: Array<String>,
    @JsonProperty("map") val mapFilePath: String,
    @JsonProperty("background") val backgroundFilePath: String?
) : Synonymized
